### PR TITLE
Add metadata to GPT

### PR DESCRIPTION
**Pull request by yuhongsun96**

### CODE REVIEW

File: backend/danswer/configs/app_configs.py [hunk 1]
```diff
@@ -96,6 +96,8 @@
 # 1 edit per 2 characters, currently unused due to fuzzy match being too slow
 QUOTE_ALLOWED_ERROR_PERCENT = 0.05
 QA_TIMEOUT = 10  # 10 seconds
+# Include additional document/chunk metadata in prompt to GenerativeAI
```
Comment (yuhongsun96) [inline backend/danswer/configs/app_configs.py:99]: doesn't work great, need prompt tuning, main problem is that GPT confuses aggregations with metadata... any questions asked on metadata it refuses to answer

File: backend/danswer/connectors/web/connector.py [hunk 2]
```diff
@@ -84,6 +85,8 @@ def load_from_state(self) -> GenerateDocumentsOutput:
             logger.info(f"Indexing {current_url}")
 
             try:
+                current_visit_time = datetime.now().strftime("%B %d, %Y, %H:%M:%S")
```
Comment (yuhongsun96) [inline backend/danswer/connectors/web/connector.py:88]: cant get page update time reliably, time visited is still relevant imo but can discuss when we add metadata to all connectors

File: backend/danswer/direct_qa/qa_prompts.py [hunk 3]
```diff
@@ -100,46 +194,6 @@ def freeform_processor(question: str, documents: list[str]) -> str:
     return prompt
 
 
-def json_chat_processor(question: str, documents: list[str]) -> list[dict[str, str]]:
```
Comment (yuhongsun96) [inline backend/danswer/direct_qa/qa_prompts.py:103]: moving it to mark the rest as deprecated, they don't perform great anyway... could revisit in the future though so not deleting

File: backend/danswer/connectors/web/connector.py [hunk 4]
```diff
@@ -104,7 +107,7 @@ def load_from_state(self) -> GenerateDocumentsOutput:
                             sections=[Section(link=current_url, text=page_text)],
                             source=DocumentSource.WEB,
                             semantic_identifier=current_url.split(".")[-1],
-                            metadata={},
+                            metadata={"time visited": current_visit_time},
```
Comment (Weves) [inline backend/danswer/connectors/web/connector.py]: nit: should we have a consistent format for the metadata? Here it's lowercase, for confluence it was uppercase (in my productboard pr it's snake case)

File: backend/danswer/direct_qa/qa_prompts.py [hunk 5]
```diff
@@ -24,20 +31,107 @@
 }
 
 
-def json_processor(question: str, documents: list[str]) -> str:
+def add_metadata_section(
+    prompt_current: str,
+    chunk: InferenceChunk,
+    prepend_tab: bool = False,
+    include_sep: bool = False,
+) -> str:
```
Comment (Weves) [inline backend/danswer/direct_qa/qa_prompts.py:39]: nit: add comment about the assumptions this function is making about `prompt_current`

**Review by Weves — APPROVED** [review]
added a few small comments to address, but overall lgtm

**Pull request merged**